responses/tests/test_responses.py

```diff
@@ -2129,3 +2129,49 @@ def run():
 
     run()
     assert_reset()
+
+
+def test_redirect():
+    @responses.activate
+    def run():
+        # create multiple Response objects where first two contain redirect headers
+        rsp1 = responses.Response(
+            responses.GET,
+            "http://example.com/1",
+            status=301,
+            headers={"Location": "http://example.com/2"},
+        )
+        rsp2 = responses.Response(
+            responses.GET,
+            "http://example.com/2",
+            status=301,
+            headers={"Location": "http://example.com/3"},
+        )
+        rsp3 = responses.Response(responses.GET, "http://example.com/3", status=200)
```

Comment:
Should we have a method for generating `ConnectionError` style responses that doesn't require getting the response value out and modifying it? Perhaps something like

```python
rsp = responses.Response(
  responses.GET,
  'http://example.com/3',
  body=requests.ConnectionError('something bad')
)
responses.add(rsp)
```

Reply:
it will be hard to achieve. With provided example we will raise an error and the object will not contain history.

thus, we first need http response (not our response) object and then we attach this object to the Exception. 
If we want to generate genuine Response, then we have to rewrite lots of things and duplicate some code from `requests` and probably apply our mock on a deeper (or maybe higher) level

Reply:
This is probably an uncommon scenario that we don't need to design an ergonomic API for. If it comes up again we can figure out a higher-level API.